### PR TITLE
PHPCS: Like module

### PR DESCRIPTION
**Pull request by sdixon194**
[(Not to be confused with **Comment** Likes Module.)](https://github.com/Automattic/jetpack/pull/19221)

This PR updates the likes module to meet phpcs standards.

#### Changes proposed in this Pull Request:

* This PR updates the likes module to meet phpcs standards.

#### Jetpack product discussion

N/a

#### Does this pull request change what data or activity we track or use?

Nope.

#### Testing instructions:

* Check out branch
* Make sure you can still enable likes under Settings > Sharing.
* Make sure you can still like posts and pages without an issue, manage email notification settings, etc.

### CODE REVIEW

File: projects/plugins/jetpack/modules/likes.php [hunk 1]
```diff
@@ -38,18 +43,19 @@ public static function init() {
...
+	/** Constructs Likes class */
+	public function __construct() {
 		$this->in_jetpack = ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ? false : true;
-		$this->settings = new Jetpack_Likes_Settings();
+		$this->settings   = new Jetpack_Likes_Settings();
 
 		// We need to run on wp hook rather than init because we check is_amp_endpoint()
-		// when bootstrapping hooks
+		// when bootstrapping hooks.
 		add_action( 'wp', array( &$this, 'action_init' ), 99 );
```
Comment (sdixon194) [inline projects/plugins/jetpack/modules/likes.php:53]: There are a few instances here where we're passing the object as reference which PHPCS doesn't like - removing the ampersand and just passing $this doesn't seem to change anything, but before I do - is there any good reason where we'd need to actually do that?

File: projects/plugins/jetpack/modules/likes.php [hunk 2]
```diff
@@ -9,6 +9,8 @@
  * Module Tags: Social
  * Feature: Engagement
  * Additional Search Queries: like, likes, wordpress.com
+ *
+ * @package automattic/Jetpack
```
Comment (kraftbj) [inline projects/plugins/jetpack/modules/likes.php:13]:
```suggestion
 * @package automattic/jetpack
```

File: projects/plugins/jetpack/modules/likes.php (hunk 1 above)
Comment (kraftbj) [reply projects/plugins/jetpack/modules/likes.php:53]: I can't think of any personally for WP hooks in this manner.

Comment (sdixon194) [reply projects/plugins/jetpack/modules/likes.php]: Removed it and it seems to be fine ¯\_(ツ)_/¯

File: projects/plugins/jetpack/modules/likes.php [hunk 3]
```diff
@@ -367,10 +381,10 @@ function load_admin_css() {
 	}
 
 	/**
-	* Load the JS required for loading the like counts.
-	*/
-	function enqueue_admin_scripts() {
-		if ( empty( $_GET['post_type'] ) || 'post' == $_GET['post_type'] || 'page' == $_GET['post_type'] ) {
```
Comment (sdixon194) [inline projects/plugins/jetpack/modules/likes.php:373]: This is the last thing that needs addressing, but it doesn't look like a nonce is set and I'm honestly not sure where in this case we'd set one.

File: projects/plugins/jetpack/modules/likes.php [hunk 4]
```diff
@@ -367,10 +381,10 @@ function load_admin_css() {
...
+	 * Load the JS required for loading the like counts.
+	 */
+	public function enqueue_admin_scripts() {
+		if ( empty( $_GET['post_type'] ) || 'post' === $_GET['post_type'] || 'page' === $_GET['post_type'] ) {
```
Comment (sdixon194) [inline projects/plugins/jetpack/modules/likes.php:387]: This line here needs nonce verification and is the last thing that needs addressing, but it doesn't look like a nonce is set and I'm honestly not sure where in this case we'd set one.

File: projects/plugins/jetpack/modules/likes.php [hunk 5]
```diff
@@ -24,12 +26,15 @@
 	'dns-prefetch'
 );
 
-include_once dirname( __FILE__ ) . '/likes/jetpack-likes-master-iframe.php';
-include_once dirname( __FILE__ ) . '/likes/jetpack-likes-settings.php';
+require_once dirname( __FILE__ ) . '/likes/jetpack-likes-master-iframe.php';
+require_once dirname( __FILE__ ) . '/likes/jetpack-likes-settings.php';
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/likes.php:30]:
```suggestion
require_once __DIR__ . '/likes/jetpack-likes-master-iframe.php';
require_once __DIR__ . '/likes/jetpack-likes-settings.php';
```
Supposed to be caught by a sniff, but upstream hasn't done a release yet to include https://phabricator.wikimedia.org/T273574.

File: projects/plugins/jetpack/modules/likes.php [hunk 6]
```diff
@@ -95,70 +101,65 @@ function __construct() {
...
-
 	/**
-     * Stub for is_post_likeable, since some wpcom functions call this directly on the class
+	 * Stub for is_post_likeable, since some wpcom functions call this directly on the class
 	 * Are likes enabled for this post?
-     *
-     * @param int $post_id
-     * @return bool
+	 *
+	 * @param int $post_id - id of the post.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/likes.php:159]: FYI, that `-` is not required (and normally not used) in PHP-land. It's only the JS eslint that requires it.

File: projects/plugins/jetpack/modules/likes.php [hunk 7]
```diff
@@ -422,49 +439,55 @@ function likes_edit_column( $column_name, $post_id ) {
...
-	* Add a "Likes" column header to the post edit table in wp-admin.
-	*
-	* @param array $columns
-	* @return array
-	*/
-	function add_like_count_column( $columns ) {
+	 * Add a "Likes" column header to the post edit table in wp-admin.
+	 *
+	 * @param array $columns - array of columns in wp-admin.
+	 * @return array $columns - column in wp-admin.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/likes.php:445]: This doesn't seen quite accurate.

There's no real need to describe the return value (phpcs doesn't insist), but if you really want to I might do something like
```suggestion
	 * @return array $columns with "likes" added.
```

File: projects/plugins/jetpack/modules/likes.php [hunk 8]
```diff
@@ -477,31 +500,40 @@ function post_likes( $content ) {
...
 
-		// Let's make sure that the script is enqueued
+		// Let's make sure that the script is enqueued.
 		wp_enqueue_script( 'jetpack_likes_queuehandler' );
 
 		return $content . $html;
 	}
 
-	function post_flair_service_enabled_like( $classes ) {
+	/** Adds sd-like-enabled CSS class?
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/likes.php:509]:
```suggestion
	/**
	 * Adds sd-like-enabled CSS class?
```

File: projects/plugins/jetpack/modules/likes.php [hunk 9]
```diff
@@ -24,12 +26,15 @@
...
 
-include_once dirname( __FILE__ ) . '/likes/jetpack-likes-master-iframe.php';
-include_once dirname( __FILE__ ) . '/likes/jetpack-likes-settings.php';
+require_once dirname( __FILE__ ) . '/likes/jetpack-likes-master-iframe.php';
+require_once dirname( __FILE__ ) . '/likes/jetpack-likes-settings.php';
 
+/** Jetpack Like Class */
 class Jetpack_Likes {
+
+	/** Initialize class */
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/likes.php:35]: While this is accepted by phpcs, personally I'd format the doc blocks consistently in long format to match those that already exist in the file.
```suggestion
	/**
	 * Initialize class
	 */
```

File: projects/plugins/jetpack/modules/likes.php (hunk 4 above)
Comment (anomiex) [reply projects/plugins/jetpack/modules/likes.php:387]: Hmm. It's not really *processing* form data anyway. It's just adding some scripts. I might just ignore this one. But you might want to ask someone more familiar with WP code than I am for confirmation.

File: projects/plugins/jetpack/modules/likes.php [hunk 10]
```diff
@@ -388,32 +403,34 @@ function enqueue_admin_scripts() {
...
+					JETPACK__VERSION,
+					$in_footer = false
 				);
 			} else {
-				wp_enqueue_script( 'jquery.wpcom-proxy-request', "/wp-content/js/jquery/jquery.wpcom-proxy-request.js", array('jquery'), NULL, true );
-				wp_enqueue_script( 'likes-post-count', plugins_url( 'likes/post-count.js', dirname( __FILE__ ) ), array( 'jquery' ), JETPACK__VERSION );
-				wp_enqueue_script( 'likes-post-count-wpcom', plugins_url( 'likes/post-count-wpcom.js', dirname( __FILE__ ) ), array( 'likes-post-count', 'jquery.wpcom-proxy-request' ), JETPACK__VERSION );
+				wp_enqueue_script( 'jquery.wpcom-proxy-request', '/wp-content/js/jquery/jquery.wpcom-proxy-request.js', array( 'jquery' ), JETPACK__VERSION, true );
+				wp_enqueue_script( 'likes-post-count', plugins_url( 'likes/post-count.js', dirname( __DIR__ ) ), array( 'jquery' ), JETPACK__VERSION, $in_footer = false );
+				wp_enqueue_script( 'likes-post-count-wpcom', plugins_url( 'likes/post-count-wpcom.js', dirname( __DIR__ ) ), array( 'likes-post-count', 'jquery.wpcom-proxy-request' ), JETPACK__VERSION, $in_footer = false );
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/likes.php:412]: Changing `dirname( __FILE__ )` to `dirname( __DIR__ )` is not equivalent. You probably meant to change to just `__DIR__`.
```suggestion
				wp_enqueue_script( 'likes-post-count', plugins_url( 'likes/post-count.js', __DIR__ ), array( 'jquery' ), JETPACK__VERSION, $in_footer = false );
				wp_enqueue_script( 'likes-post-count-wpcom', plugins_url( 'likes/post-count-wpcom.js', __DIR__ ), array( 'likes-post-count', 'jquery.wpcom-proxy-request' ), JETPACK__VERSION, $in_footer = false );
```
Or was this code broken before and you're now fixing it?

Comment (sdixon194) [reply projects/plugins/jetpack/modules/likes.php]: > You probably meant to change to just __DIR__.

Nice catch! That one was on me.

File: projects/plugins/jetpack/modules/likes.php (hunk 4 above)
Comment (sdixon194) [reply projects/plugins/jetpack/modules/likes.php:393]: Yeah, I'll cc @kraftbj here for his expert opinion.

Comment (kraftbj) [reply projects/plugins/jetpack/modules/likes.php]: Apologies for the delay, but agreed on ignoring.

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
There are a few issues on the WP.com side we should resolve. @sdixon194 Would you be game for bringing that to completion? If so, we can force approve this PR to avoid any conflicts, then get the WP.com side fixed up.

**Comment by sdixon194** [comment]
> Would you be game for bringing that to completion?

Yup, taking a look!

**Comment by kraftbj** [comment]
r224288-wpcom

**Pull request merged**